lib/install/rust.js

```diff
@@ -0,0 +1,523 @@
+        resolve(tarball);
+      });
+  });
+};
+
+module.exports.installSdk = () => {
+  exports.checkRust(false);
+
+  return exports.installTools()
+    .then(() => {
+      return exports.installRustlib()
+        .then(() => {
+          log.info('SDK installed.');
+        }, e => {
+          log.error(e.message);
+          log.warn('Please switch to using a stable Rust version >= 1.11.0.');
+          log.warn('SDK toolchain is installed, but a libstd for your Rust version is not.');
+        });
+    });
+};
+
+module.exports.uninstallSdk = () => {
+  return new Promise((resolve) => {
+    fs.remove(path.join(osenv.home(), '.tessel/rust'), () => {
+      fs.remove(path.join(osenv.home(), '.tessel/sdk'), () => {
+        log.warn('Tessel SDK uninstalled.');
+        resolve();
+      });
+    });
+  });
+};
+
+// Loging function that checks if all rust components are installed.
+module.exports.checkSdk = () => {
+  return exports.getBuildConfig()
+    .catch(e => {
+      log.error('Could not find all the components for cross-compiling Rust:');
+      log.error(e.message);
+      log.error('Please run "cargo tessel sdk install" and try again.');
+      log.error('To instead use the remote Rust compiler, use `t2 run <target> --rustcc`.');
+      process.exit(1);
+    });
+};
+
+module.exports.checkRust = (isCli) => {
+  try {
+    return exports.rustVersion();
+  } catch (e) {
+    log.error('To cross-compile Tessel code on your machine, you will need both');
+    log.error('`rustc` and `cargo` available in your terminal: https://rustup.rs/');
+    if (isCli) {
+      log.error('To instead use the remote Rust compiler, use `t2 run <target> --rustcc`.');
+    }
+    process.exit(1);
+  }
+};
+
+// Loging function that checks if rust is installed, then if the binary matches.
+module.exports.checkConfiguration = (matchBin) => {
+  var metadata = exports.cargoMetadata();
+
+  // Get first package.
+  var pkg = metadata.packages.pop();
+  var bins = pkg.targets.filter(target => target.kind.indexOf('bin') > -1);
+
+  // Filter by --bin argument.
+  var validbins = bins.filter(bin => bin.name === matchBin);
+
+  // Throw if multiple bins exist.
+  if (validbins.length === 0) {
+    if (matchBin === null) {
+      log.error(`Please specify a valid binary target with --bin.`);
+    } else {
+      log.error(`No binary target "${matchBin}" exists for this Rust crate.\nMake sure you specify a valid binary using --bin.`);
+    }
+    if (bins.length > 0) {
+      log.error('Available targets:');
+      bins.forEach(bin => {
+        log.error(' $ cargo tessel build --bin', bin.name);
+      });
+    }
+    process.exit(1);
+  }
+
+  var out = validbins[0];
+  var dest = path.join(path.dirname(pkg.manifest_path), 'target/tessel2/release', out.name);
+
+  return {
+    name: out.name,
+    path: dest
+  };
+};
+
+module.exports.runBuild = (matchBin) => {
+  return new Promise((resolve) => {
+      exports.checkRust(true);
```

Comment:
`checkRust` really needs to be a promise returning function that resolves in safe-to-proceed conditions and otherwise rejects. Since this is in the `t2 run ...` path, it must not subvert the closing process, which is must be followed to ensure the connection to board is completely closed, whether there is a complete success or complete failure.